tests/Aspire.Components.Common.Tests/ActivityNotifier.cs

```diff
@@ -20,6 +20,6 @@ public sealed class ActivityNotifier : BaseProcessor<Activity>
     public override void OnEnd(Activity data)
     {
         ExportedActivities.Add(data);
-        _taskSource.SetResult();
+        _taskSource.TrySetResult();
```

Comment:
This seems to indicate that OnEnd is getting called multiple times now. (But it wasn't before?)

This may be an issue.

Reply:
These tests should be able to be executed and debugged on your local machine. You just need Docker running so the TestContainers works.

Reply:
I haven't been able to reproduce the failure we're seeing in Helix locally. I debugged through this with @davidfowl and the `ActivityNotifier` isn't even explicitly added by the tests that are failing but this was something we saw that could conceivably contribute if tests are running in parallel.

Reply:
Looks like Linux passed this time.

Reply:
Sooo, looks like this change fixed it.

Reply:
OK we figured this out, it's an actual issue with the code change to do with a workaround to support keyed services in the Redis OTel client. Fixing.

Reply:
Doing more investigation, this was caused by a test that registered an ActivityListener in the same process as the rest of the tests. ActivityListeners are global, so they get all Activities. The Activities were never created before because the Host was never Started. With the latest changes the Activities always get created even if the Host isn't Started.

The fix is to move the ActivityListener  tests to a separate process.